Python/Product/Analysis/LanguageServer/Settings.cs

```diff
@@ -15,24 +15,62 @@
 // permissions and limitations under the License.
 
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.PythonTools.Analysis.LanguageServer {
-    internal class LanguageServerSettings {
-        public AutoCompleteSettings autoComplete;
-        public DiagnosticsSettings diagnostics;
+    public sealed class LanguageServerSettings {
+        public class PythonAnalysisOptions {
+            public bool openFilesOnly;
+            public object[] errors = Array.Empty<object>();
+            public object[] warnings = Array.Empty<object>();
+            public object[] information = Array.Empty<object>();
+            public object[] disabled = Array.Empty<object>();
```

Comment:
I'm happy to require these all be strings. LSP allows them to be numbers or strings, but I think we get to dictate which one they'll be.

Reply:
This is just for the analyzer convenience since `AddDiagnostic` et al use `object code`

Reply:
I can change that to strings if you prefer

Reply:
Those can change to strings as well

Reply:
This stems from parser and tokenizer use of `ErrorCodes` which are integers and `DiagnosticsErrorSink`. We can change `Diagnostic` to hold a string and do something like

```cs
            var d = new Diagnostic {
                code = "E{0}".FormatInvariant(errorCode),
                message = message,
                source = _source,
                severity = GetSeverity(severity),
                range = span
            };
```